dotnet/src/Connectors/Connectors.Memory.Redis/RedisMemoryStore.cs

```diff
@@ -359,6 +355,49 @@ private static RedisKey GetRedisKey(string collectionName, string key)
             timestamp: ParseTimestamp((long?)hashEntries.FirstOrDefault(x => x.Name == "timestamp").Value));
     }
 
+    private async IAsyncEnumerable<MemoryRecord> InternalGetBatchAsync(string collectionName, string[] keys, bool withEmbeddings, [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        var batch = this._database.CreateBatch();
+        var tasks = keys.Select(async key =>
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            HashEntry[] hashEntries = await this._database.HashGetAllAsync(GetRedisKey(collectionName, key), CommandFlags.None).ConfigureAwait(false);
```

Comment:
To be part of the batch I believe HashGetAllAsync needs to be called on the batch object itself.